typing/jkind.ml

```diff
@@ -1605,7 +1605,6 @@ module Builtin = struct
       { layout; mod_bounds = Bounds.max; with_bounds = No_with_bounds }
     in
     fresh_jkind_poly desc ~annotation:None ~why:(Product_creation why)
-    |> mark_best
```

Comment:
It would be good to comment why `best` is ok in `product` but not here in `product_of_sorts`. I think it's this:

```ocaml
(* [mark_best] is correct here because the with-bounds of a product jkind include all the components
of the product. Accordingly, looking through the product, by one step, never loses any information. *)
```
in `product` and then
```
(* We do not [mark_best] here because the resulting jkind is used (only) in the middle of type-checking mutually recursive type declarations. See Note [Default jkind in transl_declaration] for more commentary on why we don't want [best] jkinds there. *)
```